src/ros_filter.cpp

```diff
@@ -1541,6 +1584,14 @@ namespace RobotLocalization
 
     while (ros::ok())
     {
+      // Each second we remove old states and measurements from the queue.
+      static int flush_counter = 0;
+      flush_counter++;
+      if (flush_counter > frequency_)
+      {
+        removeHistoryStatesAndMeasurementsOlderThan(filter_.getLastMeasurementTime()-double(double(filterHistoryInterval_)/1000.0));
```

Comment:
Maybe change this to `filter_.getLastMeasurementTime() - static_cast<double>(filterHistoryInterval_) / 1000.0`. Really, the division operation will cast `filterHistoryInterval_` to a double anyway, but this is explicit and avoids the unnecessary second cast.